libs/api_common/windows_platform_common.cpp

```diff
@@ -6,60 +6,351 @@
+
+Exit:
+    return result;
+}
 
 const EbpfProgramType&
 get_program_type_windows(const GUID& program_type)
 {
-    // TODO: (Issue #67) Make an IOCTL call to fetch the program context
-    //       info and then fill the EbpfProgramType struct.
-    for (const EbpfProgramType& t : windows_program_types) {
-        if (t.platform_specific_data != (uint64_t)&EBPF_PROGRAM_TYPE_UNSPECIFIED) {
-            ebpf_program_type_t* program_type_uuid = (ebpf_program_type_t*)t.platform_specific_data;
-            if (IsEqualGUID(*program_type_uuid, program_type)) {
-                return t;
-            }
+    ebpf_result_t result;
+
+    // See if we have the descriptor in the thread local cache.
+    auto it = _program_descriptor_cache.find(program_type);
+    if (it != _program_descriptor_cache.end()) {
+        return *_program_descriptor_cache[program_type].get();
+    }
+
+    // Descriptor not found in thread local cache, try to query
+    // the info from execution context.
+    ebpf_program_info_t* program_info = nullptr;
+    EbpfProgramType* descriptor = nullptr;
+    result = _get_program_info_data(program_type, &program_info);
+    if (result == EBPF_SUCCESS) {
+        _program_info_cache[program_type] = ebpf_program_info_ptr_t(program_info);
+        // Convert program info to program descriptor.
+        result = _get_program_descriptor_from_info(program_info, &descriptor);
+        if (result == EBPF_SUCCESS) {
+            _program_descriptor_cache[program_type] = ebpf_program_descriptor_ptr_t(descriptor);
+            return *_program_descriptor_cache[program_type].get();
+        }
+    }
+
+    // Failed to query from execution context. Consult static cache.
+    if (use_ebpf_store) {
+        auto it2 = _windows_program_types.find(program_type);
+        if (it2 != _windows_program_types.end()) {
+            return it2->second;
         }
     }
 
     auto guid_string = guid_to_string(&program_type);
     throw std::runtime_error(std::string("ProgramType not found for GUID ") + guid_string);
 }
 
+static const ebpf_section_definition_t*
+_get_section_definition(const std::string& section)
+{
+    int32_t match_index = -1;
+    size_t match_length = 0;
+    for (uint32_t index = 0; index < _windows_section_definitions.size(); index++) {
+        std::string section_prefix(_windows_section_definitions[index].section_prefix);
+        if (section.find(section_prefix) == 0) {
+            size_t prefix_length = strlen(_windows_section_definitions[index].section_prefix);
+            if (match_length < prefix_length) {
+                match_index = index;
+                match_length = prefix_length;
+            }
+        }
+    }
+
+    if (match_index >= 0) {
+        return &_windows_section_definitions[match_index];
+    }
+
+    return nullptr;
+}
+
+_Ret_maybenull_ const ebpf_program_type_t*
+get_ebpf_program_type(bpf_prog_type_t bpf_program_type)
+{
+    for (auto const& [key, val] : _windows_program_information) {
```

Comment:
```suggestion
    for (auto const& [key, value] : _windows_program_information) {
```
nit: avoid abbreviations per coding style guidelines